tests/e2e/actor_features/actor_features_test.go

```diff
@@ -0,0 +1,272 @@
+
+		go func() {
+			_, err := utils.HTTPPost(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "method", "concurrency"), []byte{})
+			err1 <- err
+		}()
+		go func() {
+			_, err := utils.HTTPPost(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "method", "concurrency"), []byte{})
+			err2 <- err
+		}()
+
+		require.NoError(t, <-err1)
+		require.NoError(t, <-err2)
+
+		resp, err := utils.HTTPGet(logsURL)
+		require.NoError(t, err)
+		logOne := findNthActorAction(resp, actorID, "concurrency", 1)
+		logTwo := findNthActorAction(resp, actorID, "concurrency", 2)
+		require.True(t, (logOne != nil) && (logTwo != nil))
+		require.True(t, (logOne.StartTimestamp < logOne.EndTimestamp)) // Sanity check on the app response.
+		require.True(t, (logTwo.StartTimestamp < logTwo.EndTimestamp)) // Sanity check on the app response.
+		require.True(t, (logOne.StartTimestamp >= logTwo.EndTimestamp) || (logTwo.StartTimestamp >= logOne.EndTimestamp))
+	})
+
+	t.Run("Actor concurrency different actor ids.", func(t *testing.T) {
+		// Each test needs to have a different actorID
+		actorIDOne := "1004a"
+		actorIDTwo := "1004b"
+
+		// Invoke method call in Actors.
+		err1 := make(chan error, 1)
+		err2 := make(chan error, 1)
+
+		go func() {
+			_, err := utils.HTTPPost(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorIDOne, "method", "concurrency"), []byte{})
+			err1 <- err
+		}()
+		go func() {
+			_, err := utils.HTTPPost(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorIDTwo, "method", "concurrency"), []byte{})
+			err2 <- err
+		}()
+
+		require.NoError(t, <-err1)
+		require.NoError(t, <-err2)
+
+		resp, err := utils.HTTPGet(logsURL)
+		require.NoError(t, err)
+		logOne := findActorAction(resp, actorIDOne, "concurrency")
+		logTwo := findActorAction(resp, actorIDTwo, "concurrency")
+		require.True(t, (logOne != nil) && (logTwo != nil))
+		require.True(t, (logOne.StartTimestamp < logOne.EndTimestamp)) // Sanity check on the app response.
+		require.True(t, (logTwo.StartTimestamp < logTwo.EndTimestamp)) // Sanity check on the app response.
+		// Both methods run in parallel, with the sleep time both should start before the other ends.
+		require.True(t, (logOne.StartTimestamp <= logTwo.EndTimestamp) && (logTwo.StartTimestamp <= logOne.EndTimestamp))
+	})
+
+	t.Run("Actor failsover to another hostname.", func(t *testing.T) {
+		// Each test needs to have a different actorID
+		actorID := "1005"
+
+		// In Kubernetes, hostname should be the POD name. Single-node Kubernetes cluster should still be able to reproduce this test.
+		firstHostname, err := utils.HTTPPost(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "method", "hostname"), []byte{})
+		require.NoError(t, err)
+
+		tr.Platform.Restart(appName)
+
+		newHostname, err := utils.HTTPPost(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "method", "hostname"), []byte{})
+		require.NoError(t, err)
+
+		require.True(t, string(firstHostname) != string(newHostname))
```

Comment:
nit: We can use `require.Equal(t, firstHostname, newHostname)`

Reply:
Done.